### PR TITLE
Rl 14+15

### PR DESCRIPTION
**Pull request by Squidtito**
I've added a description and inventory system. Descriptions have been added to the existing items in the game.

### CODE REVIEW

File: scripts/inventory.gd [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		itemlist.remove_item(itemlist.get_selected_items()[0])
+	
+
+func additem(id):
+	if items.has(id) == true:
+		itemlist.add_item(id, load(items.get(id).picture))
+	
+func _on_ItemList_item_selected(index):
+	$CanvasLayer/ColorRect/RichTextLabel.text = items.get(itemlist.get_item_text(index)).description
+	pass
```
Comment (keri4141) [inline scripts/inventory.gd]: can you remove this since it's not necessary

File: scripts/inventory.gd [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"description": 'The most powerful knife',
+	},
+	"Wooden Knife": {
+	"description": 'A knife made out of wood, not very effective...',
+	'picture': 'res://assets/items/wooden_knife.png',
+	}
+}
+func _ready():
+	$CanvasLayer.visible = false #<---- I can't do this in editor, so I have to do it via code.
+	additem('Knife')
```
Comment (keri4141) [inline scripts/inventory.gd:19]: I think doing a loop here might be better, than manually adding each item.

Hint: You'll have to first get the list of keys from the dictionary and loop through those.

Comment (keri4141) [inline scripts/inventory.gd]: I think doing a loop here might be better, than manually adding each item.

Hint: You'll have to first get the list of keys from the dictionary and loop through those.

File: scripts/inventory.gd [hunk 3]
```diff
@@ -0,0 +1,44 @@
+extends Control
+onready var itemlist:ItemList = $CanvasLayer/ItemList
+var items:Dictionary = {
+	"Knife": {
+	'picture': 'res://assets/items/knife.png',
```
Comment (keri4141) [inline scripts/inventory.gd]: can you indent the fields inside the Knife object

File: scripts/inventory.gd [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	"description": 'A standard knife',
```
Comment (keri4141) [inline scripts/inventory.gd]: indent this as well

File: scripts/inventory.gd [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	},
+	"Hell Knife": {
+	'picture': 'res://assets/items/hell_knife.png',
```
Comment (keri4141) [inline scripts/inventory.gd]: same here

File: scripts/inventory.gd [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	"description": 'The most powerful knife',
+	},
+	"Wooden Knife": {
+	"description": 'A knife made out of wood, not very effective...',
```
Comment (keri4141) [inline scripts/inventory.gd]: same here

File: scripts/inventory.gd [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+	if Input.is_action_just_pressed("inventory"):
+		if $CanvasLayer.visible == false:
+			$CanvasLayer.visible = true
+		else:
+			$CanvasLayer.visible = false
+			$CanvasLayer/ColorRect/RichTextLabel.text = 'Select an item'
+
+func _on_Button_pressed():
+	if itemlist.get_selected_items().size() >= 1:
+		itemlist.remove_item(itemlist.get_selected_items()[0])
```
Comment (keri4141) [inline scripts/inventory.gd]: itemlist.get_selected_items()[0]

is a bit awkward, yeah it works...but for readability can we do something like

var selectedItems= itemlist.get_selected_items()
itemlist.remove_item(selectedItems[0])

**Review by keri4141 — CHANGES_REQUESTED** [review]
capitalize scenes please.

inventory.tscn
player.tscn

are not capitalized

**Comment by Squidtito** [comment]
I have commited the fixes for these issues.

File: scripts/inventory.gd [hunk 8]
```diff
@@ -2,23 +2,24 @@ extends Control
...
+		'picture': 'res://assets/items/wooden_knife.png',
 	}
 }
 func _ready():
 	$CanvasLayer.visible = false #<---- I can't do this in editor, so I have to do it via code.
-	additem('Knife')
-	additem('Wooden Knife')
-	additem('Hell Knife')
+	var count = 0
+	while not count == 3:
```
Comment (keri4141) [inline scripts/inventory.gd]: Are we only going to have 3 items or were we planning to have more? What if we wanted to add more? And if we did add more we would have to keep coming here to modify this code right? Is there a better way of doing this?

Comment (Squidtito) [reply scripts/inventory.gd]: There are gonna only be 3 item slots, and this is just placeholder code, so I mean... You're not gonna have the strongest knife in the game when you first start it lol

File: scripts/inventory.gd [hunk 9]
```diff
@@ -32,7 +33,8 @@ func _process(_delta):
 
 func _on_Button_pressed():
 	if itemlist.get_selected_items().size() >= 1:
-		itemlist.remove_item(itemlist.get_selected_items()[0])
+		var selectedItems= itemlist.get_selected_items()
```
Comment (keri4141) [inline scripts/inventory.gd]: add a space here before the '='

Comment (Squidtito) [reply scripts/inventory.gd]: Fixed

File: cvsd.gd [hunk 10]
```diff
@@ -0,0 +1,16 @@
+extends Node
```
Comment (keri4141) [inline cvsd.gd]: is this cvsd.gd file suppose to be here? Is it being used?

Comment (Squidtito) [reply cvsd.gd]: Oh sorry, I don't know how that got there, fixed

**Review by keri4141 — APPROVED** [review]

**Pull request merged**